src/components/ItemList/ItemList.tsx

```diff
@@ -13,31 +14,57 @@ export type Props<T extends { id: string }> = {
 
 const ItemList = <T extends { id: string }>({
     activeItemIndex = 0,
+    autoScroll = false,
     itemRowAs: ItemRowComponent = ItemRow,
     items,
     onActivate = noop,
     onSelect,
     ...rest
-}: Props<T>): JSX.Element => (
-    <ul className="ba-ItemList" data-testid="ba-ItemList" role="listbox" {...rest}>
-        {items.map((item, index) => (
-            <ItemRowComponent
-                key={item.id}
-                className="ba-ItemList-row"
-                isActive={index === activeItemIndex}
-                item={item}
-                onClick={(event: React.SyntheticEvent) => {
-                    onSelect(index, event);
-                }}
-                onMouseDown={(event: React.SyntheticEvent) => {
-                    event.preventDefault(); // preventDefault on mousedown so blur doesn't happen before click
-                }}
-                onMouseEnter={() => {
-                    onActivate(index);
-                }}
-            />
-        ))}
-    </ul>
-);
+}: Props<T>): JSX.Element => {
+    const listRef = React.useRef<HTMLUListElement>(null);
+
+    const activeRef = React.useCallback(
+        (activeEl: ItemRowRef) => {
+            const { current: listEl } = listRef;
+
+            if (!autoScroll || !activeEl || !listEl) {
+                return;
+            }
+
+            const { offsetHeight: activeHeight, offsetTop: activeTop } = activeEl;
+            const { offsetHeight: listHeight, scrollTop: listTop } = listEl;
+
+            if (listTop > activeTop) {
```

Comment:
Would `scroll-into-view-if-needed` be useful here? I see it as a dependency for annotations but we don't use it anywhere currently

Reply:
Thank you for pointing it out. I will take a look